src/test/java/io/cdap/plugin/servicenow/source/ServiceNowSourceTest.java

```diff
@@ -16,108 +16,234 @@
-      Assert.assertEquals("Errors were encountered during validation.",
-                          e.getMessage());
+      Assert.assertEquals("Table: sys_user is empty.", e.getFailures().get(0).getMessage());
     }
   }
 
   @Test
-  public void testPrepareRun() {
+  public void testPrepareRun() throws Exception {
     MockFailureCollector mockFailureCollector = new MockFailureCollector();
     MockArguments mockArguments = new MockArguments();
     BatchSourceContext context = Mockito.mock(BatchSourceContext.class);
     Mockito.when(context.getFailureCollector()).thenReturn(mockFailureCollector);
     Mockito.when(context.getArguments()).thenReturn(mockArguments);
+    ServiceNowTableAPIClientImpl restApi = Mockito.mock(ServiceNowTableAPIClientImpl.class);
+    PowerMockito.whenNew(ServiceNowTableAPIClientImpl.class).
+      withArguments(Mockito.any(ServiceNowBaseSourceConfig.class)).thenReturn(restApi);
+    List<Map<String, Object>> result = new ArrayList<>();
+    Map<String, Object> map = new HashMap<>();
+    map.put("key", "value");
+    result.add(map);
+    int httpStatus = HttpStatus.SC_OK;
+    Map<String, String> headers = new HashMap<>();
+    String responseBody = "{\n" +
+      "    \"result\": []\n" +
+      "}";
+    RestAPIResponse restAPIResponse = new RestAPIResponse(httpStatus, headers, responseBody);
+    Mockito.when(restApi.executeGet(Mockito.any())).thenReturn(restAPIResponse);
+    Mockito.when(restApi.parseResponseToResultListOfMap(restAPIResponse.getResponseBody())).thenReturn(result);
+    OAuthClient oAuthClient = Mockito.mock(OAuthClient.class);
+    PowerMockito.whenNew(OAuthClient.class).
+      withArguments(Mockito.any(URLConnectionClient.class)).thenReturn(oAuthClient);
+    OAuthJSONAccessTokenResponse accessTokenResponse = Mockito.mock(OAuthJSONAccessTokenResponse.class);
+    Mockito.when(oAuthClient.accessToken(Mockito.any(), Mockito.anyString(), Mockito.any(Class.class))).
+      thenReturn(accessTokenResponse);
+    Mockito.when(accessTokenResponse.getAccessToken()).thenReturn("token");
+    RestAPIResponse response = Mockito.spy(restAPIResponse);
+    CloseableHttpClient httpClient = Mockito.mock(CloseableHttpClient.class);
+    HttpClientBuilder httpClientBuilder = Mockito.mock(HttpClientBuilder.class);
+    PowerMockito.mockStatic(HttpClientBuilder.class);
+    PowerMockito.mockStatic(RestAPIResponse.class);
+    PowerMockito.when(HttpClientBuilder.create()).thenReturn(httpClientBuilder);
+    Mockito.when(httpClientBuilder.build()).thenReturn(httpClient);
+    CloseableHttpResponse httpResponse = Mockito.mock(CloseableHttpResponse.class);
+    Mockito.when(httpClient.execute(Mockito.any())).thenReturn(httpResponse);
+    PowerMockito.when(RestAPIResponse.parse(ArgumentMatchers.any(), ArgumentMatchers.anyString())).
+      thenReturn(response);
     serviceNowSource.prepareRun(context);
     Assert.assertEquals(0, mockFailureCollector.getValidationFailures().size());
```

Comment:
Same comment as above

Reply:
Added another assertion for checking whether the `context` has a non-null `multisink.table name` argument value. c0e6a0f